### DISCUSSION TITLE
Decoding Postgres numeric to JS number

### DISCUSSION THREAD

**Discussion by linhub15** [denodrivers/postgres#414] [Q&A] [answered]
Is there a particular reason that the Postgres data type of `numeric` is returned as a string instead of a JavaScript `number`?

The postgres numeric does potentially hold a massive number with lots more precision than a JS number, but in most use cases wouldn't it be okay?

https://github.com/denodrivers/postgres/blob/main/query/decode.ts#L74-L92

**Comment by bombillazo** [comment]
Im also interested in this. its not only numeric but also Floats (even float4 which technically fits in JS float) are all parsed as text

**Comment by bombillazo** [comment, marked as answer]
After we stabilize the project dependencies, we'll look into how to enhance the experience to customize and give devs more controls over the transformations done to/from the DB, I too find myself overriding many of the hardcoded mappings for the DB data column types to JS types.

**Comment by bombillazo** [comment]
A PR has been opened to allow devs to control this behavior #458
